aptible/client_utils.go

```diff
@@ -41,31 +42,40 @@ func generateErrorFromClientError(abstractedError interface{}) error {
 		return fmt.Errorf("Unable to properly decode error in unmarshal from client - %s\n", err.Error())
 	}
 
-	if out.Payload.Code == nil || out.Payload.Error == nil {
-		var errString string
+	// payload is a pointer and can be nil, but we don't have all our information from a http code, so we construct it
+	if out.Payload == nil {
+		errorString = fmt.Sprint(
+			"Error without a valid payload: ", abstractedError, "\n",
+		)
+	} else if out.Payload.Code == nil || out.Payload.Error == nil {
 		if err != nil {
-			errString = fmt.Sprintf("- %s", err.Error())
+			errorString = fmt.Sprintf("- %s", err.Error())
 		}
 		if out.Payload.Code != nil && *out.Payload.Code >= 400 {
-			errString += fmt.Sprintf(" status code (%d)", *out.Payload.Code)
+			errorString += fmt.Sprintf(" status code (%d)", *out.Payload.Code)
 		}
+
 		if out.Payload.Error != nil && len(*out.Payload.Error) > 0 {
-			errString += fmt.Sprintf(" error (%s)", *out.Payload.Error)
+			errorString += fmt.Sprintf(" error (%s)", *out.Payload.Error)
 		}
+
 		if out.Payload.Message != nil && len(*out.Payload.Message) > 0 {
-			errString += fmt.Sprintf(" error message (%s)", *out.Payload.Message)
+			errorString += fmt.Sprintf(" error message (%s)", *out.Payload.Message)
 		}
+
 		return fmt.Errorf(
-			"unable to properly decode error (missing fields to properly generate error)%s",
-			errString,
+			"unable to properly decode error (missing fields to properly generate error) - %s\n",
+			errorString,
```

Comment:
It's possible that `errorString` hasn't been set here but `var errorString string` defaults to the zero value of empty string so it won't fail, correct?

Reply:
Also, what's the difference between returning the error here vs setting the error string and letting it return at the end of the method?